src/main/java/edu/byu/cs/controller/ConfigController.java

```diff
@@ -124,18 +179,30 @@ private static JsonObject getPrivateConfig() throws DataAccessException {
         dao.setConfiguration(ConfigurationDao.Configuration.BANNER_MESSAGE, message, String.class);
         dao.setConfiguration(ConfigurationDao.Configuration.BANNER_LINK, link, String.class);
         dao.setConfiguration(ConfigurationDao.Configuration.BANNER_COLOR, color, String.class);
+        dao.setConfiguration(ConfigurationDao.Configuration.BANNER_EXPIRATION, expirationTimestamp, Instant.class);
 
         User user = req.session().attribute("user");
         if (message.isEmpty()) {
             logConfigChange("cleared the banner message", user.netId());
         } else {
-            logConfigChange("set the banner message to: '%s' with link: {%s}".formatted(message, link), user.netId());
+            expirationString = !expirationString.isEmpty() ? expirationString : "never";
+            logConfigChange("set the banner message to: '%s' with link: {%s} to expire at %s".formatted(message, link, expirationString), user.netId());
         }
 
         res.status(200);
         return "";
     };
 
+    private static Instant getInstantFromUnzonedTime(String timestampString) throws DateTimeParseException {
+        ZoneId utahZone = ZoneId.of("America/Denver");
+
+        DateTimeFormatter formatter = DateTimeFormatter.ISO_LOCAL_DATE_TIME;
+        LocalDateTime localDateTime = LocalDateTime.parse(timestampString, formatter);
+        ZonedDateTime utahTime = localDateTime.atZone(utahZone);
+
+        return utahTime.toInstant();
+    }
```

Comment:
Do we want to be hardcoding this? @frozenfrank what do you think?

Reply:
Let's save this improvement in #156.